numcodecs/fletcher32.pyx

```diff
@@ -0,0 +1,45 @@
+import struct
+
+from numcodecs.abc import Codec
+from numcodecs.compat import ensure_contiguous_ndarray
+
+from libc.stdint cimport uint8_t, uint16_t, uint32_t
```

Comment:
Does this need to be in Cython? It seems most of the functionality here can be done in Python. If the C file moves to Cython, this should be doable in pure Python. This should simplify things significantly from a maintenance standpoint. Things like inheriting from Python classes in Cython can sometimes be wonky